aiohomekit/controller/ble/pairing.py

```diff
@@ -596,20 +746,79 @@ async def _populate_accessories_and_characteristics(
             if config_changed:
                 self._callback_and_save_config_changed(self.config_num)
 
+            if not was_connected:
+                logger.debug(
+                    "%s: Connected, processing subscriptions: %s; rssi=%s",
+                    self.name,
+                    self.subscriptions,
+                    self.rssi,
+                )
+                # Only start active subscriptions if we stay connected for more
+                # than subscription delay seconds.
+                self._restore_subscriptions_timer = asyncio.get_event_loop().call_later(
+                    SUBSCRIPTION_RESTORE_DELAY, self._restore_subscriptions
+                )
+
     def _restore_subscriptions(self):
         """Restore subscriptions after after connecting."""
         if self.client and self.client.is_connected:
-            async_create_task(
-                self._async_start_notify_subscriptions(list(self.subscriptions))
-            )
+            async_create_task(self._async_restore_subscriptions())
 
-    async def _async_start_notify_subscriptions(
+    async def _async_subscribe_broadcast_events(
         self, subscriptions: list[tuple[int, int]]
     ) -> None:
-        """Start notifications for the given subscriptions."""
-        if not self.accessories or not self.client.is_connected:
+        """Subscribe to broadcast events."""
+        accessory_chars = self.accessories.aid(1).characteristics
+        to_subscribe: list[Characteristic] = []
+        for _, iid in subscriptions:
+            hap_char = accessory_chars.iid(iid)
+            if not hap_char:
+                continue
+            if not hap_char.broadcast_events:
+                continue
+            if iid in self._broadcast_notifications:
+                continue
+            to_subscribe.append(hap_char)
+
+        if not to_subscribe:
             return
 
+        async with self._ble_request_lock:
+            for hap_char in to_subscribe:
+                iid = hap_char.iid
+                if iid in self._broadcast_notifications:
+                    continue  # check again with the lock
+                logger.debug(
+                    "%s: Subscribing to broadcast notify for iid: %s", self.name, iid
+                )
+                enable_broadcast_payload = b"\x01\x02\x01\x00\x02\x01\x01"
```

Comment:
We don't have TLVs for this so its hand rolled, it would be nice to add the constants

Reply:
done